cmd/bancheck/config/config.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+)
+
+type BannedApi struct {
+	Name       string      `json:"name"` // fully qualified identifier name
+	Msg        string      `json:"msg"`  // additional information e.g. rationale for banning
+	Exemptions []Exemption `json:"exemptions"`
+}
+
+// BannedApis is a map of identifier names to a list of all corresponding
+// BannedApi entries with additional information.
+type BannedApis map[string][]BannedApi
+
+type Exemption struct {
+	Justification string `json:"justification"`
+	AllowedDir    string `json:"allowedDir"`
+}
+
+// ReadBannedImports reads banned imports from all config files
+// and concatenates them into one object.
+func ReadBannedImports(files []string) (BannedApis, error) {
```

Comment:
I decided to split Read() into ReadBannedImports() and ReadBannedFunctions(). I think that makes more sense because the client will only be interested in one or the other, not both.

Reply:
> the client will only be interested in one or the other, not both.

Why? Who is the client in your mind?

Reply:
The [analyzers](https://pkg.go.dev/golang.org/x/tools/go/analysis#hdr-Analyzer), one analyzer will check for banned imports and the other for banned function calls.

Reply:
I think this is one of the benefits of splitting config-reading logic from actual usage of these -- that you *don't* need to split reading "different parts of a config".

I'd leave it as it was before and do a single `ReadConfig` func and apply all my comments from the other PR.

As it is right now:
- this is basically a copy paste, including all the boilerplate related to reading a file and so on
- you read a file multiple times; not a big issue per se (we don't need to worry about performance)

Reply:
Makes sense. I changed it back.